lib/tests/common.go

```diff
@@ -0,0 +1,144 @@
+package test
+
+import (
+	"archive/tar"
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/appc/docker2aci/lib/internal/typesV2"
+)
+
+type Layer map[*tar.Header][]byte
```

Comment:
Ah, I overlooked you're trying to use stdlib's `tar.Header` as the key type. http://godoc.org/archive/tar#Header contains `Xattrs` of type `map[string]string` and that is the problem.

Map [key types](https://golang.org/ref/spec#KeyType) must have the comparison operators == and != fully defined, hence your struct key type must not contain a function, map, or slice.

I agree that this is used only in tests, but if you'll ever put two equal header values, but with different memory locations, they won't be the same :-/